getting-started.md

```diff
@@ -35,406 +38,669 @@ While this getting started guide is meant to be self-contained, you may also fin
+    Currently, this is just ignoring the frequency specification for inductor, since that data isn't available in the parts table.
+- `compile_board_inplace(...)` invokes the circuit generator given the top-level design (`BlinkyExample`).
+  This is the starting point that allows the file to run as a Python script, and you can treat it as magic.
+
+Try building the example now:
+- **If not using the IDE**: run `python blinkly_skeleton.py` from the command line.
+  If all worked, this should create a folder `PolymorphicNlocks/blinky_skeleton` with a netlist `BlinkyExample.net` inside.
+- **If using the IDE**: look for the run icon ![run icon](docs/ide/ide_run_button.png) in the gutter (with the line numbers) next to `class BlinkyExample`.
+  1. Click it.
+     _Make sure that you're using the run icon associated with `class BlinkyExample`, not the file, and not `if __name__ == "__main__"`._
+  2. Then from the menu, click the Run option.  
+     ![run menu](docs/ide/ide_run_blinky_menu.png)
+     > Next time, you can rebuild the design by re-running the last selected run configuration with hotkey **Shift+F10**.
+  3. The design should build, and you should get a run log that looks something like:
+     ```
+     Starting compilation of blinky_skeleton.BlinkyExample
+     Using interpreter from configured SDK [...]
+     [... lots of compilation output here ...]
+     Run backend NetlistBackend
+     Wrote netlist to [...]
+     ```
+  4. Unlike the command-line option, this generates a netlist in the same directory as `blinky_skeleton.py`.
+     In the IDE, you can configure where this goes via the run options at the top right:  
+     ![run menu](docs/ide/ide_run_config.png)
+
 
 ### Creating the microcontroller and LED
-For this simple example, we will use a socketed development board, a Nucleo F303K8.
-It's very convenient and simple since it integrates a programming connection (via USB), provides power to the rest of your circuit (from USB), and includes a reasonably powerful microcontroller. 
+For this simple example, we connect an LED to a STM32F103 microcontroller, and have everything powered by a USB type-C receptacle.
 
-**In `# implementation here`, add this code** to instantiate the microcontroller board / socket and LED as follows:
+**In `blinky_skeleton.py`, `# you implementation here`, add this code** to instantiate the microcontroller and LED as follows:
 ```python
-self.mcu = self.Block(Nucleo_F303k8())
+self.usb = self.Block(UsbCReceptacle())
+self.mcu = self.Block(Stm32f103_48())
 self.led = self.Block(IndicatorLed())
 ```
 > `self.Block(...)` creates a sub-block in `self` (the current hierarchy block being defined).
 > It must be assigned to an instance variable (in this case, `mcu`), which is used as the name sub-block.
 
+> You can also insert blocks through graphical operations in the IDE.
+> 1. Start by selecting the insert location in code, by setting the caret to the end of `super().contents()`.
```

Comment:
Maybe we can emphasize and explain that the caret must be at that specific position.